add release notes and version numbers for v0.5.5

AdamISZ opened the pull request:
Opened PR so people can read the notes in advance of finalising the release.

undeath commented:
I think it would be helpful to add the new cgan onion address to the release notes (and mention it in the `upgrading` section).

Everything else looks good to me.

AdamISZ commented:
Good idea, will do.

kristapsk commented:
Did some manual testing and also running yg for a few days already with this, haven't noticed any new issues.

AlexCato commented:
I know @AdamISZ already confirmed he saw it on IRC, but just that it's noted here as well: found a serious bug during manual testing, which makes sendpayment-sweeps impossible on the current master branch. Fix must be included in the upcoming release: #382 . Kinda stupid mistake I introduced, this should be something that a regression test needs to find in the future. I'll look into test writing as soon as I find some time for that.

YG tests: all good, no problems found
sendpayment-tests: except the issue above, nothing else found

AdamISZ commented:
Updates: added onion, added new commits for txfee check from @AlexCato but not merging this, used cherry-pick instead (because I prefer one clean commit for the release tag rather than a merge). Hence closing this. Thanks for review.
